src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.cs

```diff
@@ -1258,14 +1258,22 @@ private async Task<TokenValidationResult> ValidateTokenAsync(JsonWebToken jsonWe
                         }
                     }
 
-                    if (TokenUtilities.IsRecoverableConfiguration(validationParameters, currentConfiguration, out currentConfiguration))
+                    if (TokenUtilities.ShouldValidateWithLKG(validationParameters))
```

Comment:
If we make the change in BaseConfigurationManager.GetValidLkgConfiguraitons(), TokenUtilities.ShouldValidateWithLKG is not needed as we assume all entries in the lkgCache should be tried.
If there are none, an empty array will be returned.